.github/workflows/gradle.yml

```diff
@@ -164,6 +164,25 @@ jobs:
           command: SUB_BUILD=PLATFORM ./gradlew --no-daemon assemble --scan
           attempt_limit: 3
           attempt_delay: 5000 # in ms
+      
+      - name: Build Python Package
```

Comment:
The publication to PyPI should happen when a new version is released, not on build.
I think it belongs to this [workflow](https://github.com/airbytehq/airbyte/blob/f94ef99f3d436434903f8c1357898b39423eaeab/.github/workflows/release-airbyte-os.yml#L72) instead. 
Moreover, these GitHub secrets are not defined on our repo so it won't work out of the box. 
Let's reuse the same action structure we use to publish the CDK ([here](https://github.com/airbytehq/airbyte/blob/master/.github/workflows/publish-cdk-command-manually.yml#L158) cc @maxi297 ) the action is different (mariamrf/py-package-publish-action@v1.1.0).

Reply:
These suggestions have been implemented since this comment. @alafanechere any further review?